### PR TITLE
infer host interface from default route

### PR DESCRIPTION
**Pull request by ibotty**

### CODE REVIEW

**Comment by ibotty** [comment]
fixes #5

**Comment by ibotty** [comment]
There could be more automagic :unicorn::  Say if the IP is set, try to find the corresponding interface.  I don't know, whether that's really useful though, because if you explicitly set the IP you might be able to set the interface as well.

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -19,9 +19,27 @@ if [ ! -z "${KARP_DEBUG}" ] ; then
   /usr/sbin/ucarp --help
 fi
 
+# if no explicit host interface is set, we determine it from the default route
+if [ -z "$KARP_INTERFACE" ] ; then
+  KARP_INTERFACE="$(ip r | awk '/^default / { for(i=1; i<=NF; i++) if ($i == "dev") print $(i+1) }')"
```
Comment (immanuelfodor) [inline docker-entrypoint.sh:24]: Nice `awk` trick! :)

**Review by immanuelfodor — APPROVED** [review]

**Comment by immanuelfodor** [comment]
Released in: https://github.com/immanuelfodor/kube-karp/releases/tag/1.0.3

**Pull request merged**